sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClientBuilder.java

```diff
@@ -202,6 +207,14 @@ private HttpPipeline getHttpPipeline() {
 
         policies.add(new AddDatePolicy());
 
+        // We need to place this policy right before the credential policy since headers may affect the string to sign
```

Comment:
I would agree with this assessment, it would be possible to add `x-ms-*` or other headers involved in the string to sign.